src/app/src/pages/redteam/setup/components/Targets/tabs/ResponseTransformTab.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+
+import Box from '@mui/material/Box';
+import { useTheme } from '@mui/material/styles';
+import Typography from '@mui/material/Typography';
+import dedent from 'dedent';
+import Prism from 'prismjs';
+import Editor from 'react-simple-code-editor';
+import type { ProviderOptions } from '@promptfoo/types';
+
+interface ResponseTransformTabProps {
+  selectedTarget: ProviderOptions;
+  updateCustomTarget: (field: string, value: any) => void;
+}
+
+const highlightJS = (code: string): string => {
+  try {
+    const grammar = (Prism as any)?.languages?.javascript;
+    if (!grammar) {
+      return code;
+    }
+    return Prism.highlight(code, grammar, 'javascript');
+  } catch {
+    return code;
+  }
+};
+
+const ResponseTransformTab: React.FC<ResponseTransformTabProps> = ({
+  selectedTarget,
+  updateCustomTarget,
+}) => {
+  const theme = useTheme();
+  const darkMode = theme.palette.mode === 'dark';
+
+  return (
+    <>
+      <Typography variant="body1" sx={{ mb: 2 }}>
+        Extract specific data from the HTTP response. See{' '}
+        <a href="https://www.promptfoo.dev/docs/providers/http/#response-transform" target="_blank">
+          docs
+        </a>{' '}
+        for more information.
+      </Typography>
+      <Box
+        sx={{
+          border: 1,
+          borderColor: 'grey.300',
+          borderRadius: 1,
+          position: 'relative',
+          backgroundColor: darkMode ? '#1e1e1e' : '#fff',
+        }}
+      >
+        <Editor
+          value={selectedTarget.config.transformResponse || ''}
+          onValueChange={(code) => updateCustomTarget('transformResponse', code)}
+          highlight={highlightJS}
+          padding={10}
+          placeholder={dedent`Optional: Transform the API response before using it. Format as either:
```

Comment:
It would be great if you can take the raw response from target testing, parse it, generate a type hint, and display that here.  That would be a lot more helpful than the placeholder which people may find confusing e.g. when `choices` isn't included in their response.